garage/np/algos/base.py

```diff
@@ -1,7 +1,91 @@
-class Algorithm:
-    pass
+from collections import deque
 
+from dowel import tabular
+import numpy as np
 
-class RLAlgorithm(Algorithm):
-    def train(self):
-        raise NotImplementedError
+from garage.misc import special, tensor_utils
+
+
+class RLAlgorithm():
+    def __init__(self,
```

Comment:
i don't think we want a constructor or methods on `RLAlgorithm` -- it should only be an interface class. 

plus, lots of these constructor arguments and process_samples looks very specific to BatchPolopt.

@naeioi please advise

Reply:
@zequnyu You dont need the parentheses.